### DISCUSSION TITLE
Apache airflow 2.3.3 : How to manually trigger airflow scheduler  so that it can close and reopen it's log file ?

### DISCUSSION THREAD

**Discussion by nikhilnitrr** [apache/airflow#29722] [Q&A]
Hi,
I have a requirement of airflow scheduler log rotation. Is there a way to trigger airflow scheduler PID with any kind of signal (like SIGUSR1, SIGHUP....etc) so that it can just close and reopen it's log file without causing any affect to any task.

Please help me, I referred airflow documentation by didn't get any useful information related to my use case.

**Comment by potiuk** [comment]
No. There is no way to do it. But this is not the right way of doing it either.

Airflow uses standard Python logging framework and you should simply implement the configuration that will do what you want as described in https://airflow.apache.org/docs/apache-airflow/stable/administration-and-deployment/logging-monitoring/logging-tasks.html#advanced-configuration and use rotating file handler in place of the default "console" handler. We even have our own RotatingFileHandler implemented for [DAG File Processor.](https://github.com/apache/airflow/blob/main/airflow/config_templates/airflow_local_settings.py#L154) 

That requires a bit of playing with your local settings and you can get log rotation implemented as you want - for those logs and components that you want. See
